### PR TITLE
[Bugfix] Mv not refresh after truncate base table

### LINKED ISSUE #9931: [sync materialized view]not refresh after truncate base table
### Steps to reproduce the behavior (Required)
```
CREATE TABLE `t2` (
                                            `k1` date,
                                            `k2` int,
                                            `k3` int
                                        ) ENGINE=OLAP
                                        DUPLICATE KEY(`k1`)
                                        COMMENT "OLAP"
                                        DISTRIBUTED BY HASH(`k1`) BUCKETS 3;
INSERT INTO t2 VALUES ("2020-11-10",3,3);
CREATE MATERIALIZED VIEW mv1
                             DISTRIBUTED BY HASH(k1) BUCKETS 10
                             REFRESH ASYNC
                             AS SELECT k1,k2 from t2;
TRUNCATE TABLE t2;
select * from mv1;
```
### Expected behavior (Required)
`Empty set (0.01 sec)`
### Real behavior (Required)
```
+------------+------+
| k1         | k2   |
+------------+------+
| 2020-11-10 |    3 |
+------------+------+
```
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
```
+----------------------------+
| current_version()          |
+----------------------------+
| BRANCH-2.4-RELEASE f823e8b |
+----------------------------+
```

### PR DESCRIPTION
**Pull request by blackstar-baba**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #9931

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 7 / 9 (77.78%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|7|9|77.78%|[4209, 4210]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 7 / 9 (77.78%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|7|9|77.78%|[4209, 4210]|

File: fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java [hunk 1]
```diff
@@ -4193,9 +4193,21 @@ public void truncateTable(TruncateTableStmt truncateTableStmt) throws DdlExcepti
...
+                if (materializedView.isLoadTriggeredRefresh()) {
+                    refreshMaterializedView(db.getFullName(), db.getTable(mvId).getName(),
+                            Constants.TaskRunPriority.NORMAL.value());
+                }
+            }
         } catch (DdlException e) {
             deleteUselessTabletAndShard(tabletIdSet, copiedTbl);
             throw e;
+        } catch (MetaNotFoundException e) {
+            LOG.warn("Table {} related materialized view can not find", e);
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java:4210]:
```suggestion
            LOG.warn("Table related materialized view can not be found", e);
```

File: fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java [hunk 2]
```diff
@@ -4193,9 +4193,21 @@ public void truncateTable(TruncateTableStmt truncateTableStmt) throws DdlExcepti
             TruncateTableInfo info = new TruncateTableInfo(db.getId(), olapTable.getId(), newPartitions,
                     truncateEntireTable);
             editLog.logTruncateTable(info);
+
+            // refresh mv
+            Set<Long> relatedMvs = olapTable.getRelatedMaterializedViews();
+            for (long mvId : relatedMvs) {
+                MaterializedView materializedView = (MaterializedView) db.getTable(mvId);
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java:4200]: can materializedView be null?

**Review by Astralidea — DISMISSED** [review]

**Review by ABingHuang — CHANGES_REQUESTED** [review]

File: fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java (hunk 2 above)
Comment (blackstar-baba) [reply fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java:4200]: I think mv can not be null here because we use write lock wrap this logic

**Comment by Astralidea** [comment]
run starrocks_fe_unittest

**Comment by Astralidea** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 7 / 9 (77.78%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|7|9|77.78%|[4220, 4221]|

**Review by Astralidea — APPROVED** [review]

**Review by ABingHuang — APPROVED** [review]
LGTM

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by gengjun-git — APPROVED** [review]

**Comment by Astralidea** [comment]
@Mergifyio backport branch-2.4

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 7 / 9 (77.78%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|7|9|77.78%|[4220, 4221]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 7 / 9 (77.78%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|7|9|77.78%|[4220, 4221]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 7 / 9 (77.78%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|7|9|77.78%|[4220, 4221]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 7 / 9 (77.78%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|7|9|77.78%|[4220, 4221]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 7 / 9 (77.78%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|7|9|77.78%|[4220, 4221]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 7 / 9 (77.78%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|7|9|77.78%|[4220, 4221]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 7 / 9 (77.78%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|7|9|77.78%|[4220, 4221]|

**Pull request merged**